clang/unittests/Format/FormatTestComments.cpp

```diff
@@ -796,6 +796,44 @@ TEST_F(FormatTestComments, ParsesCommentsAdjacentToPPDirectives) {
             format("namespace {}\n   /* Test */    #define A"));
 }
 
+TEST_F(FormatTestComments, DeIdentsCommentBeforeIfdefAfterBracelessIf) {
+  verifyFormat("void f() {\n"
+               "  if (true)\n"
+               "    int i;\n"
+               "  /* comment */\n"
+               "#ifdef A\n"
+               "  int j;\n"
+               "#endif\n"
+               "}",
+               "void f() {\n"
+               "  if (true)\n"
+               "    int i;\n"
+               "    /* comment */\n"
+               "#ifdef A\n"
+               "  int j;\n"
+               "#endif\n"
+               "}");
+
+  verifyFormat("void f() {\n"
+               "  if (true)\n"
+               "    int i;\n"
+               "  /* comment */\n"
```

Comment:
It doesn't look right as the comment is likely for the line above it.

Reply:
Okay, I don't think that is the existing behavior. Using `clang-format-18`
```
void f() {
  if (foo) 
    a = 3;
    /* comment */

  int b = 4; 
}
```
is formatted to
```
void f() {
  if (foo)
    a = 3;
  /* comment */

  int b = 4;
}
```